### ISSUE TITLE
[BUG] oiiotool unusable on Apple Silicon

### ISSUE THREAD

**Issue by carlocab** [OpenImageIO/oiio#2858]
**Describe the bug**
Using `oiiotool` after building natively on Apple Silicon produces the following error:
```
==> /opt/homebrew/Cellar/openimageio/2.2.11.1/bin/oiiotool --info /opt/homebrew/Library/Homebrew/test/support/fixtures/test.jpg 2>&1
oiiotool ERROR: read : "/opt/homebrew/Library/Homebrew/test/support/fixtures/test.jpg": unknown error
Full command line was:
> oiiotool --info /opt/homebrew/Library/Homebrew/test/support/fixtures/test.jpg
```

**To Reproduce**
1. Build OIIO natively on Apple Silicon
2. Run `oiiotool --info` on any jpeg file.

**Expected behavior**
No errors.

**Evidence**
This error was encountered while trying to package OIIO for Apple Silicon: https://github.com/Homebrew/homebrew-core/pull/70469

CI logs: https://github.com/Homebrew/homebrew-core/pull/70469/checks?check_run_id=1835730809

**Platform information:**
 - OIIO branch/version: 2.2.11.1
 - OS: macOS 11.2
 - C++ compiler: Apple clang version 12.0.0 (clang-1200.0.32.29)
 - Any non-default build flags when you build OIIO: None.

**IF YOU ALREADY HAVE A CODE FIX:** There is no need to file a separate issue,
please just go straight to making a [pull request](https://github.com/OpenImageIO/oiio/pulls).

**Comment by carlocab** [comment]
I don't have an M1 machine myself, so it's a bit difficult for me to get proper debug information. However, if you have a preferred method for extracting debug information that can be scripted, I'll attempt to run it on Homebrew CI.

**Comment by carlocab** [comment]
Strangely enough, this failure seems to depend on the path of the jpeg file you try to pass to `oiiotool`. See https://github.com/Homebrew/homebrew-core/pull/72567 for details.

**Comment by lgritz** [comment]
I don't have an M1, either. What I really need is for somebody with an M1 who can interactively reproduce this error, and then we can have a debugging correspondence session and get to the bottom of it, with me putting code on GH with some extra error instrumentation that can narrow this down.

**Comment by cho-m** [comment]
I'll update this thread with whatever findings I have and I'll try getting output you request whenever I have free cycles to spend on this.

Here is latest stacktrace after building as Debug:
```console
❯ /opt/homebrew/bin/oiiotool --info test12345678901234.jpg
test12345678901234.jpg :    1 x    1, 3 channel, uint8 jpeg

❯ /opt/homebrew/bin/oiiotool --info test123456789012345.jpg
/tmp/openimageio-20210603-27391-guutbe/oiio-Release-2.2.15.0/src/libutil/ustring.cpp:396: TableRep: Assertion 'str.c_str() == c_str() && str.size() == length' failed.
 0# OpenImageIO_v2_2::Sysutil::stacktrace() in /opt/homebrew/Cellar/openimageio/2.2.15.0/lib/libOpenImageIO.2.2.15.dylib
 1# OpenImageIO_v2_2::stacktrace_signal_handler(int) in /opt/homebrew/Cellar/openimageio/2.2.15.0/lib/libOpenImageIO.2.2.15.dylib
 2# _sigtramp in /usr/lib/system/libsystem_platform.dylib
 3# pthread_kill in /usr/lib/system/libsystem_pthread.dylib
 4# abort in /usr/lib/system/libsystem_c.dylib
 5# OpenImageIO_v2_2::ustring::TableRep::TableRep(OpenImageIO_v2_2::string_view, unsigned long) in /opt/homebrew/Cellar/openimageio/2.2.15.0/lib/libOpenImageIO.2.2.15.dylib
 6# OpenImageIO_v2_2::TableRepMap<256u, 4096u>::insert(OpenImageIO_v2_2::string_view, unsigned long) in /opt/homebrew/Cellar/openimageio/2.2.15.0/lib/libOpenImageIO.2.2.15.dylib
 7# OpenImageIO_v2_2::ustring::make_unique(OpenImageIO_v2_2::string_view) in /opt/homebrew/Cellar/openimageio/2.2.15.0/lib/libOpenImageIO.2.2.15.dylib
 8# input_file(int, char const**) in /opt/homebrew/bin/oiiotool
 9# std::__1::function<void (OpenImageIO_v2_2::ArgParse::Arg&, OpenImageIO_v2_2::span<char const* const, -1l>)>::operator()(OpenImageIO_v2_2::ArgParse::Arg&, OpenImageIO_v2_2::span<char const* const, -1l>) const in /opt/homebrew/Cellar/openimageio/2.2.15.0/lib/libOpenImageIO.2.2.15.dylib
10# OpenImageIO_v2_2::ArgParse::Impl::parse_args(int, char const**) in /opt/homebrew/Cellar/openimageio/2.2.15.0/lib/libOpenImageIO.2.2.15.dylib
11# OpenImageIO_v2_2::ArgParse::parse_args(int, char const**) in /opt/homebrew/Cellar/openimageio/2.2.15.0/lib/libOpenImageIO.2.2.15.dylib
12# getargs(int, char**) in /opt/homebrew/bin/oiiotool
13# main in /opt/homebrew/bin/oiiotool
zsh: abort      /opt/homebrew/bin/oiiotool --info test123456789012345.jpg
```

**Comment by carlocab** [comment]
CC @cho-m, who's been trying to debug this. They've figured out the error arises when the path passed to oiiotool is too long.

**Comment by lgritz** [comment]
oh, that's a really interesting stack trace.

That seems really different from the "oiiotool ERROR: read :" errors you were getting before.

Are you saying that there are some times when it gives you an error message, and other times when it has a legit crash?

I have some ideas to explore, but I really need somebody who can try things out an an M1 for me.

This kind of thing is going to be a lot easier when GitHub offers M1 as a choice for Actions CI.

**Comment by cho-m** [comment]
In Homebrew, we will build as Release, so I think all the errors weren't showing up.

I am currently running a build with `-DCMAKE_BUILD_TYPE=Debug` and this is stacktrace that occurred.

I can try things out as I am using an M1 mac.

**Comment by lgritz** [comment]
oh, I see -- you're saying you get a crash in debug mode, but a vague error in release mode?

Are you able to build from source if I tell you a git branch to try?

Just to test this, are you able to build the current 'master' and get the crash and/or error? If so, I'll make a test branch for you today that will help us narrow this down.

**Comment by cho-m** [comment]
I did see similar error after building master branch, though I can't guarantee all the build flags are the same from original environment.

Even `oiiotool --help` has similar stacktrace.

I am assuming that Release build should remove assertions, which may be why we aren't seeing exception in Homebrew's CI.

---

Also, here is local `make` output for `ustring.cpp` where the assertion is triggered:

`cd /Users/cho-m/Desktop/oiio/build/src/libutil && /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++ -DOpenImageIO_EXPORTS -DOpenImageIO_Util_EXPORTS -DUSE_FREETYPE=1 -DUSE_OCIO=1 -DUSE_OPENCOLORIO=1 -D__STDC_CONSTANT_MACROS -D__STDC_LIMIT_MACROS -I/Users/cho-m/Desktop/oiio/build/include/OpenImageIO -I/Users/cho-m/Desktop/oiio/build/include -I/Users/cho-m/Desktop/oiio/build/src/include -I/Users/cho-m/Desktop/oiio/src/include -I/Users/cho-m/Desktop/oiio/ext/fmt/include -I/usr/local/include -isystem /opt/homebrew/include -isystem /opt/homebrew/include/Imath -g -arch arm64 -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk -fPIC -Wall -Werror -Wno-unused-function -Wno-overloaded-virtual -Wno-unneeded-internal-declaration -Wno-unused-private-field -Wno-tautological-compare -Qunused-arguments -Wunknown-warning-option -Wno-unused-local-typedefs -fno-math-errno -std=c++14 -MD -MT src/libutil/CMakeFiles/OpenImageIO_Util.dir/ustring.cpp.o -MF CMakeFiles/OpenImageIO_Util.dir/ustring.cpp.o.d -o CMakeFiles/OpenImageIO_Util.dir/ustring.cpp.o -c /Users/cho-m/Desktop/oiio/src/libutil/ustring.cpp`

**Comment by lgritz** [comment]
In ustring.cpp, on line 379, you'll see

    #elif defined(_LIBCPP_VERSION)

can you try changing that to

    #elif defined(_LIBCPP_VERSION) && !_LIBCPP_BIG_ENDIAN

??? 

I have a feeling that my handling of that case for big endian turned out not to be correct and perhaps was never tested on a real machine.

**Comment by cho-m** [comment]
Tried modifying and re-compiling, but still same error.


<details>
<summary>Local Build stacktrace for <tt>--info</tt></summary>

```cpp
❯ ../bin/oiiotool --info test123456789012345.jpg
/Users/cho-m/Desktop/oiio/src/libutil/ustring.cpp:396: TableRep: Assertion 'str.c_str() == c_str() && str.size() == length' failed.
 0# OpenImageIO_v2_3_5::Sysutil::stacktrace() in /Users/cho-m/Desktop/oiio/build/lib/libOpenImageIO_Util_d.2.3.5.dylib
 1# OpenImageIO_v2_3_5::stacktrace_signal_handler(int) in /Users/cho-m/Desktop/oiio/build/lib/libOpenImageIO_Util_d.2.3.5.dylib
 2# _sigtramp in /usr/lib/system/libsystem_platform.dylib
 3# pthread_kill in /usr/lib/system/libsystem_pthread.dylib
 4# abort in /usr/lib/system/libsystem_c.dylib
 5# OpenImageIO_v2_3_5::ustring::TableRep::TableRep(OpenImageIO_v2_3_5::string_view, unsigned long) in /Users/cho-m/Desktop/oiio/build/lib/libOpenImageIO_Util_d.2.3.5.dylib
 6# OpenImageIO_v2_3_5::ustring::TableRep::TableRep(OpenImageIO_v2_3_5::string_view, unsigned long) in /Users/cho-m/Desktop/oiio/build/lib/libOpenImageIO_Util_d.2.3.5.dylib
 7# OpenImageIO_v2_3_5::TableRepMap<256u, 4096u>::make_rep(OpenImageIO_v2_3_5::string_view, unsigned long) in /Users/cho-m/Desktop/oiio/build/lib/libOpenImageIO_Util_d.2.3.5.dylib
 8# OpenImageIO_v2_3_5::TableRepMap<256u, 4096u>::insert(OpenImageIO_v2_3_5::string_view, unsigned long) in /Users/cho-m/Desktop/oiio/build/lib/libOpenImageIO_Util_d.2.3.5.dylib
 9# OpenImageIO_v2_3_5::UstringTable::insert(OpenImageIO_v2_3_5::string_view, unsigned long) in /Users/cho-m/Desktop/oiio/build/lib/libOpenImageIO_Util_d.2.3.5.dylib
10# OpenImageIO_v2_3_5::ustring::make_unique(OpenImageIO_v2_3_5::string_view) in /Users/cho-m/Desktop/oiio/build/lib/libOpenImageIO_Util_d.2.3.5.dylib
11# OpenImageIO_v2_3_5::ustring::ustring(OpenImageIO_v2_3_5::string_view) in /Users/cho-m/Desktop/oiio/build/test/../bin/oiiotool
12# OpenImageIO_v2_3_5::ustring::ustring(OpenImageIO_v2_3_5::string_view) in /Users/cho-m/Desktop/oiio/build/test/../bin/oiiotool
13# input_file(int, char const**) in /Users/cho-m/Desktop/oiio/build/test/../bin/oiiotool
14# OpenImageIO_v2_3_5::ArgParse::Arg::action(int (*)(int, char const**))::'lambda'(OpenImageIO_v2_3_5::ArgParse::Arg&, OpenImageIO_v2_3_5::span<char const* const, -1l>)::operator()(OpenImageIO_v2_3_5::ArgParse::Arg&, OpenImageIO_v2_3_5::span<char const* const, -1l>) const in /Users/cho-m/Desktop/oiio/build/test/../bin/oiiotool
15# decltype(std::__1::forward<OpenImageIO_v2_3_5::ArgParse::Arg::action(int (*)(int, char const**))::'lambda'(OpenImageIO_v2_3_5::ArgParse::Arg&, OpenImageIO_v2_3_5::span<char const* const, -1l>)&>(fp)(std::__1::forward<OpenImageIO_v2_3_5::ArgParse::Arg&>(fp0), std::__1::forward<OpenImageIO_v2_3_5::span<char const* const, -1l> >(fp0))) std::__1::__invoke<OpenImageIO_v2_3_5::ArgParse::Arg::action(int (*)(int, char const**))::'lambda'(OpenImageIO_v2_3_5::ArgParse::Arg&, OpenImageIO_v2_3_5::span<char const* const, -1l>)&, OpenImageIO_v2_3_5::ArgParse::Arg&, OpenImageIO_v2_3_5::span<char const* const, -1l> >(OpenImageIO_v2_3_5::ArgParse::Arg::action(int (*)(int, char const**))::'lambda'(OpenImageIO_v2_3_5::ArgParse::Arg&, OpenImageIO_v2_3_5::span<char const* const, -1l>)&, OpenImageIO_v2_3_5::ArgParse::Arg&, OpenImageIO_v2_3_5::span<char const* const, -1l>&&) in /Users/cho-m/Desktop/oiio/build/test/../bin/oiiotool
16# void std::__1::__invoke_void_return_wrapper<void>::__call<OpenImageIO_v2_3_5::ArgParse::Arg::action(int (*)(int, char const**))::'lambda'(OpenImageIO_v2_3_5::ArgParse::Arg&, OpenImageIO_v2_3_5::span<char const* const, -1l>)&, OpenImageIO_v2_3_5::ArgParse::Arg&, OpenImageIO_v2_3_5::span<char const* const, -1l> >(OpenImageIO_v2_3_5::ArgParse::Arg::action(int (*)(int, char const**))::'lambda'(OpenImageIO_v2_3_5::ArgParse::Arg&, OpenImageIO_v2_3_5::span<char const* const, -1l>)&, OpenImageIO_v2_3_5::ArgParse::Arg&, OpenImageIO_v2_3_5::span<char const* const, -1l>&&) in /Users/cho-m/Desktop/oiio/build/test/../bin/oiiotool
17# std::__1::__function::__alloc_func<OpenImageIO_v2_3_5::ArgParse::Arg::action(int (*)(int, char const**))::'lambda'(OpenImageIO_v2_3_5::ArgParse::Arg&, OpenImageIO_v2_3_5::span<char const* const, -1l>), std::__1::allocator<OpenImageIO_v2_3_5::ArgParse::Arg::action(int (*)(int, char const**))::'lambda'(OpenImageIO_v2_3_5::ArgParse::Arg&, OpenImageIO_v2_3_5::span<char const* const, -1l>)>, void (OpenImageIO_v2_3_5::ArgParse::Arg&, OpenImageIO_v2_3_5::span<char const* const, -1l>)>::operator()(OpenImageIO_v2_3_5::ArgParse::Arg&, OpenImageIO_v2_3_5::span<char const* const, -1l>&&) in /Users/cho-m/Desktop/oiio/build/test/../bin/oiiotool
18# std::__1::__function::__func<OpenImageIO_v2_3_5::ArgParse::Arg::action(int (*)(int, char const**))::'lambda'(OpenImageIO_v2_3_5::ArgParse::Arg&, OpenImageIO_v2_3_5::span<char const* const, -1l>), std::__1::allocator<OpenImageIO_v2_3_5::ArgParse::Arg::action(int (*)(int, char const**))::'lambda'(OpenImageIO_v2_3_5::ArgParse::Arg&, OpenImageIO_v2_3_5::span<char const* const, -1l>)>, void (OpenImageIO_v2_3_5::ArgParse::Arg&, OpenImageIO_v2_3_5::span<char const* const, -1l>)>::operator()(OpenImageIO_v2_3_5::ArgParse::Arg&, OpenImageIO_v2_3_5::span<char const* const, -1l>&&) in /Users/cho-m/Desktop/oiio/build/test/../bin/oiiotool
19# std::__1::__function::__value_func<void (OpenImageIO_v2_3_5::ArgParse::Arg&, OpenImageIO_v2_3_5::span<char const* const, -1l>)>::operator()(OpenImageIO_v2_3_5::ArgParse::Arg&, OpenImageIO_v2_3_5::span<char const* const, -1l>&&) const in /Users/cho-m/Desktop/oiio/build/lib/libOpenImageIO_Util_d.2.3.5.dylib
20# std::__1::function<void (OpenImageIO_v2_3_5::ArgParse::Arg&, OpenImageIO_v2_3_5::span<char const* const, -1l>)>::operator()(OpenImageIO_v2_3_5::ArgParse::Arg&, OpenImageIO_v2_3_5::span<char const* const, -1l>) const in /Users/cho-m/Desktop/oiio/build/lib/libOpenImageIO_Util_d.2.3.5.dylib
21# OpenImageIO_v2_3_5::ArgParse::Impl::parse_args(int, char const**) in /Users/cho-m/Desktop/oiio/build/lib/libOpenImageIO_Util_d.2.3.5.dylib
22# OpenImageIO_v2_3_5::ArgParse::parse_args(int, char const**) in /Users/cho-m/Desktop/oiio/build/lib/libOpenImageIO_Util_d.2.3.5.dylib
23# getargs(int, char**) in /Users/cho-m/Desktop/oiio/build/test/../bin/oiiotool
24# main in /Users/cho-m/Desktop/oiio/build/test/../bin/oiiotool
zsh: abort      ../bin/oiiotool --info test123456789012345.jpg
```
</details>

<details>
<summary>Local Build stacktrace for <tt>--help</tt></summary>

```cpp
❯ bin/oiiotool --help
oiiotool -- simple image processing operations
OpenImageIO 2.3.5dev http://www.openimageio.org

...

    --unpremult                   Divide all color channels of the current image by the alpha
                                    to "un-premultiply"
    --premult                     Multiply all color channels of the current image by the alpha

/Users/cho-m/Desktop/oiio/src/libutil/ustring.cpp:396: TableRep: Assertion 'str.c_str() == c_str() && str.size() == length' failed.
 0# OpenImageIO_v2_3_5::Sysutil::stacktrace() in /Users/cho-m/Desktop/oiio/build/lib/libOpenImageIO_Util_d.2.3.5.dylib
 1# OpenImageIO_v2_3_5::stacktrace_signal_handler(int) in /Users/cho-m/Desktop/oiio/build/lib/libOpenImageIO_Util_d.2.3.5.dylib
 2# _sigtramp in /usr/lib/system/libsystem_platform.dylib
 3# pthread_kill in /usr/lib/system/libsystem_pthread.dylib
 4# abort in /usr/lib/system/libsystem_c.dylib
 5# OpenImageIO_v2_3_5::ustring::TableRep::TableRep(OpenImageIO_v2_3_5::string_view, unsigned long) in /Users/cho-m/Desktop/oiio/build/lib/libOpenImageIO_Util_d.2.3.5.dylib
 6# OpenImageIO_v2_3_5::ustring::TableRep::TableRep(OpenImageIO_v2_3_5::string_view, unsigned long) in /Users/cho-m/Desktop/oiio/build/lib/libOpenImageIO_Util_d.2.3.5.dylib
 7# OpenImageIO_v2_3_5::TableRepMap<256u, 4096u>::make_rep(OpenImageIO_v2_3_5::string_view, unsigned long) in /Users/cho-m/Desktop/oiio/build/lib/libOpenImageIO_Util_d.2.3.5.dylib
 8# OpenImageIO_v2_3_5::TableRepMap<256u, 4096u>::insert(OpenImageIO_v2_3_5::string_view, unsigned long) in /Users/cho-m/Desktop/oiio/build/lib/libOpenImageIO_Util_d.2.3.5.dylib
 9# OpenImageIO_v2_3_5::UstringTable::insert(OpenImageIO_v2_3_5::string_view, unsigned long) in /Users/cho-m/Desktop/oiio/build/lib/libOpenImageIO_Util_d.2.3.5.dylib
10# OpenImageIO_v2_3_5::ustring::make_unique(OpenImageIO_v2_3_5::string_view) in /Users/cho-m/Desktop/oiio/build/lib/libOpenImageIO_Util_d.2.3.5.dylib
11# OpenImageIO_v2_3_5::ustring::ustring(OpenImageIO_v2_3_5::string_view) in /Users/cho-m/Desktop/oiio/build/lib/libOpenImageIO_d.2.3.5.dylib
12# OpenImageIO_v2_3_5::ustring::ustring(OpenImageIO_v2_3_5::string_view) in /Users/cho-m/Desktop/oiio/build/lib/libOpenImageIO_d.2.3.5.dylib
13# OpenImageIO_v2_3_5::getattribute(OpenImageIO_v2_3_5::string_view, OpenImageIO_v2_3_5::TypeDesc, void*) in /Users/cho-m/Desktop/oiio/build/lib/libOpenImageIO_d.2.3.5.dylib
14# OpenImageIO_v2_3_5::get_string_attribute(OpenImageIO_v2_3_5::string_view, OpenImageIO_v2_3_5::string_view) in /Users/cho-m/Desktop/oiio/build/bin/oiiotool
15# formatted_format_list(OpenImageIO_v2_3_5::string_view, OpenImageIO_v2_3_5::string_view) in /Users/cho-m/Desktop/oiio/build/bin/oiiotool
16# print_help_end(std::__1::basic_ostream<char, std::__1::char_traits<char> >&) in /Users/cho-m/Desktop/oiio/build/bin/oiiotool
17# print_help(OpenImageIO_v2_3_5::ArgParse&) in /Users/cho-m/Desktop/oiio/build/bin/oiiotool
18# getargs(int, char**) in /Users/cho-m/Desktop/oiio/build/bin/oiiotool
19# main in /Users/cho-m/Desktop/oiio/build/bin/oiiotool
zsh: abort      bin/oiiotool --help
```
</details>

Diff:
```diff
diff --git a/src/libutil/ustring.cpp b/src/libutil/ustring.cpp
index 7f6df36f..d7594092 100644
--- a/src/libutil/ustring.cpp
+++ b/src/libutil/ustring.cpp
@@ -376,7 +376,7 @@ ustring::TableRep::TableRep(string_view strref, size_t hash)
     OIIO_DASSERT(str.c_str() == c_str() && str.size() == length);
     return;

-#elif defined(_LIBCPP_VERSION)
+#elif defined(_LIBCPP_VERSION) && !_LIBCPP_BIG_ENDIAN
     // libc++ uses a different std::string representation than gcc.  For
     // long char sequences, it's two size_t's (capacity & length) followed
     // by the pointer to allocated characters. (Gory detail: see the
```

**Comment by lgritz** [comment]
OK, just as a test for now, change the same line to

    #elif defined(_LIBCPP_VERSION) && 0

I just want to verify that if we don't do the code in that block, everything else seems to be ok.

Do you know what preprocessor symbol we can check to know if this is being compiled for an M1 Mac versus an Intel?

**Comment by cho-m** [comment]
I did try just removing the entire section and didn't hit the assertion error, which I think would be similar to what you are looking for.

Not exactly sure what is recommended preprocessor symbols.
Perhaps some of the arch-related ones like `__arm64__` or `__aarch64__` for M1 mac's ARM processor and `__x86_64__` for Intel processor (optionally combined with `__APPLE__`)

**Comment by lgritz** [comment]
OK, I will propose a fix based on __arch64__ that will skip the problem code. 

When I get more hands-on access to an M1, I'll figure out what was really going wrong and how to fix the broken code. Disabling it makes the ustrings internally take more memory, but will be correct and is worth the tradeoff until I can find the root problem.

**Comment by cho-m** [comment]
Thanks, @lgritz
Once there is a commit available, we can try applying patch on Homebrew-side and make it available to users.

**Comment by lgritz** [comment]
Here's the PR: https://github.com/OpenImageIO/oiio/pull/2990

If you verify that this works, I will backport and release an OIIO 2.2.15.1 with just this one little fix.

**Comment by cho-m** [comment]
Previous issues are gone with patch:
```console
❯ file bin/oiiotool
bin/oiiotool: Mach-O 64-bit executable arm64

❯ bin/oiiotool --info /opt/homebrew/Library/Homebrew/test/support/fixtures/test.jpg
/opt/homebrew/Library/Homebrew/test/support/fixtures/test.jpg :    1 x    1, 3 channel, uint8 jpeg

❯ bin/oiiotool --help
oiiotool -- simple image processing operations
OpenImageIO 2.3.5dev http://www.openimageio.org
...
Input formats supported: bmp, cineon, dds, dpx, ffmpeg, fits, gif, hdr, heif, ico, iff, jpeg, null, openexr, png, pnm, psd, raw, rla, sgi, socket,
    softimage, targa, tiff, webp, zfile
Output formats supported: bmp, dpx, fits, gif, hdr, heif, ico, iff, jpeg, null, openexr, png, pnm, rla, sgi, socket, targa, term, tiff, webp, zfile
OpenColorIO 2.0.1, color config: built-in
Known color spaces: "linear", "default", "rgb", "RGB", "sRGB", "Rec709"
Filters available: box, triangle, gaussian, sharp-gaussian, catmull-rom, blackman-harris, sinc, lanczos3, radial-lanczos3, nuke-lanczos6, mitchell,
    bspline, disk, cubic, keys, simon, rifman
Dependent libraries: FFMpeg Lavf58.76.100, gif_lib 5.2.1, libheif 1.12.0, jpeglib 9.4, null 1.0, OpenEXR 3.0.3, libpng 1.6.37, libraw 0.20.2-Release,
    LIBTIFF Version 4.3.0, Webp 1.2.0
OIIO 2.3.5dev built for C++14/201402 no SIMD
Running on 8 cores 16.0GB
Full OIIO documentation can be found at
    https://openimageio.readthedocs.io
```

Is there a specific test you want me to run?

**Comment by lgritz** [comment]
I'm sure that's fine. I'll merge the fix and cut a release when CI is done.

**Comment by lgritz** [comment]
Backported and released as v2.2.15.1

**Comment by carlocab** [comment]
Thanks, @lgritz.
